src/index.ts

```diff
@@ -74,57 +138,76 @@ import { IconAddBorder, IconStretch, IconAddBackground, IconPicture } from '@cod
  *                           also can contain any additional data that will be saved and passed back
  * @property {string} file.url - [Required] image source URL
  */
-export default class ImageTool {
+interface UploadResponseFormat {
   /**
-   * Notify core that read-only mode is supported
-   *
-   * @returns {boolean}
+   * success - 1 for successful uploading, 0 for failure 
    */
-  static get isReadOnlySupported() {
-    return true;
-  }
-
+  success: number;
   /**
-   * Get Tool toolbox settings
-   * icon - Tool icon's SVG
-   * title - title to show in toolbox
-   *
-   * @returns {{icon: string, title: string}}
+   * Object with file data.
+   *             'url' is required,
+   *             also can contain any additional data that will be saved and passed back
    */
-  static get toolbox() {
-    return {
-      icon: IconPicture,
-      title: 'Image',
-    };
-  }
+  file: {
+    /**
+     * The URL of the uploaded image.
+     */
+    url: string;
+  };
+}
 
+interface BlockToolConstructorOptions {
```

Comment:
lets use `BlockToolConstructorOptions` type from editor package
```suggestion
type ImageToolConstructorOptions = BlockToolConstructorOptions<ImageToolData, ImageToolConfig>
```